pkg/nrtupdater/nrtupdater.go

```diff
@@ -130,6 +134,63 @@ func (te *NRTUpdater) updateNRTInfo(nrt *v1alpha2.NodeResourceTopology, info Mon
 	nrt.Attributes = info.Attributes.DeepCopy()
 	nrt.Attributes = append(nrt.Attributes, te.makeAttributes()...)
 	// TODO: check for duplicate attributes?
+
+	te.updateOwnerReferences(nrt)
+}
+
+// updateOwnerReferences ensure nrt.OwnerReferences include a reference to the Node with the same name as the NRT
+//
+// Check nrt.OwnerReferences for Node references and update it so it has only one Node reference,
+// the one to the Node with the same name as the NRT.
+func (te *NRTUpdater) updateOwnerReferences(nrt *v1alpha2.NodeResourceTopology) {
+	node, err := te.k8sInterface.CoreV1().Nodes().Get(context.TODO(), nrt.Name, metav1.GetOptions{})
+	if err != nil {
+		klog.V(2).Infof("nrtupdater unable to get Node %s. Can't add Owner reference. error: %v", nrt.Name, err)
+		return
+	}
+	nodeReference := metav1.OwnerReference{
+		APIVersion: node.APIVersion,
+		Kind:       node.Kind,
+		Name:       node.Name,
+		UID:        node.UID,
+	}
+
+	// look for all the OwnerReferences pointing to a Node
+	nodeOwnerReferences := []metav1.OwnerReference{}
+	for _, own := range nrt.OwnerReferences {
+		if own.Kind == "Node" {
+			nodeOwnerReferences = append(nodeOwnerReferences, own)
+		}
+	}
+	numnodeowners := len(nodeOwnerReferences)
```

Comment:
RTE owns the NRT objects so I think you can just force the OwnerReference field the way you want. No need to be so nice (also applies to the lines below)

Reply:
Ok,  updated in the new commit also.